### ISSUE TITLE
Major bug in ajax add_to_cart method

### ISSUE THREAD

**Issue by aslamdoctor** [woocommerce/woocommerce#29893]
Filename: [woocommerce/includes/class-wc-ajax.php
](https://github.com/woocommerce/woocommerce/blob/trunk/includes/class-wc-ajax.php)

Line no **416 to 420**

Code:

```php
if ( $product && 'variation' === $product->get_type() ) {
	$variation_id = $product_id;
	$product_id   = $product->get_parent_id();
	$variation    = $product->get_variation_attributes();
}
```

1. The first line where it says '**variation**', it is supposed to be '**variable**'.
2. The line where it says `$product_id   = $product->get_parent_id();` return 0 as $product_id
3. The line where it says `$product->get_variation_attributes();` throws and error as there is no such method for $product object.

I was trying to implement ajax based add to cart feature from loop where I simply passed data attributes in the link for :

```
data-quantity
data-product_id
data-variation_id
data-attribute_ATTR_NAME
```
And it never add that specific variation in the cart because of this bug.

Looking forward to get it fixed.

**Comment by juliaamosova** [comment]
Hi @aslamdoctor,

Thank you for opening the issue! It requires further feedback from the WooCommerce Core team. I am adding the `needs developer feedback` label to this issue so that the Core team could take a look. 

Please note it may take a few days for them to get to this issue. Thank you for your patience. 

Noting here also that https://github.com/woocommerce/woocommerce/issues/29875 could be related but I couldn't reproduce it.

**Comment by barryhughes** [comment]
Hi @aslamdoctor,

[Variable products](https://docs.woocommerce.com/document/variable-product/) do indeed have a type of [`'variable'`](https://github.com/woocommerce/woocommerce/blob/5.3.0/includes/class-wc-product-variable.php#L44-L46), however the code you referenced would seem to be written in such a way that it is looking out for [product variations](https://docs.woocommerce.com/document/variable-product/#section-4) (which have a type of [`'variation'`](https://github.com/woocommerce/woocommerce/blob/5.3.0/includes/class-wc-product-variation.php#L72-L74)). The way those things are named sometimes gets a little confusing, but the first (variable products) refers the parent product and the second refers to the individual variants within those products.

Put another way, it sounds like perhaps you are trying to pass in the ID of a _variable product,_ whereas the code in question is expecting the ID of an _individual variant._ With that in mind, I don't believe there is a bug here and will close the issue.

If you disagree, please let me know and if you can provide a bit more detail on what you're doing and how your code looks, that would be great. Thanks!
